lib/registration.pm

```diff
@@ -325,18 +325,23 @@ sub fill_in_registration_data {
                     # go to the top of the list before looking for the addon
                     send_key "home";
                     # move the list of addons down until the current addon is found
-                    if (check_var('VERSION', '12-SP4') && $addon =~ 'phub') {
-                        record_soft_failure 'bsc#1092568';
+                    if ($addon eq 'phub') {
+                        # Record soft-failure for 12-SP4 and 15-SP1
+                        my $bugref = is_sle('=12-SP4') ? 'bsc#1092568' :
```

Comment:
trinary operator inside trinary operator looks too much as for me. 
what about :
```
if(is_sle('=12-SP4')) {
record_soft_failure 'bsc#1092568';
next;
} elif ( is_sle('=15-SP1') {
record_soft_failure 'bsc#1106085' ;
next;
} 
```
dramatically less code but paying with duplication lines ( which not ALWAYS evil  IMHO ... )

or if you want to "optimize"
```
my $bugref;
if(is_sle('=12-SP4')) {
$bugref = 'bsc#1092568';
} elif ( is_sle('=15-SP1') {
$bugref = 'bsc#1106085' ;
} 
if ($bugref) {
record_soft_failure $bugref;
 next;
 }
```
more code to read which is higher price than have two duplicate lines IMHO.
But both cases are better trinary operator inside trinary operator. But it is just IMHO :)

Reply:
Using multi-condition ternary operators is a common practice, and with proper formatting it's as readable as if statements. I've missed new line symbol, so fixed that now. Other than that it's a matter of taste and I believe it's pretty readable. As it's not urgent PR, it's interesting to get more opinions on that. Thanks for the feedback!

Also see: http://igoro.com/archive/a-neat-way-to-express-multi-clause-if-statements-in-c-based-languages/